internal/usecases/pegin/recommended_pegin.go

```diff
@@ -0,0 +1,202 @@
+	scaledBase := new(big.Int).Mul(big.NewInt(useCase.scale), result)
+	result.Quo(scaledBase, totalPercentages)
+
+	if err = useCase.validateRecommendedValue(ctx, config, result); err != nil {
+		return usecases.RecommendedOperationResult{}, err
+	}
+
+	return usecases.RecommendedOperationResult{
+		RecommendedQuoteValue: entities.NewBigWei(result),
+		EstimatedGasFee:       entities.NewBigWei(gasFeeEstimation),
+		EstimatedCallFee: entities.NewBigWei(
+			useCase.estimateCallFee(result, fixedCallFeeEstimation, scaledCallFeePercentage),
+		),
+		EstimatedProductFee: entities.NewBigWei(
+			useCase.estimateProductFee(result, scaledProductFee),
+		),
+	}, nil
+}
+
+func (useCase *RecommendedPeginUseCase) getScaledCallFeePercentage(
+	config liquidity_provider.PeginConfiguration,
+) *big.Int {
+	var scaledPercentageFee = new(big.Int)
+	new(big.Float).Mul(
+		new(big.Float).Quo(
+			config.FeePercentage.Native(),
+			utils.NewBigFloat64(100).Native(),
+		),
+		big.NewFloat(float64(useCase.scale)),
+	).Int(scaledPercentageFee)
+
+	return scaledPercentageFee
+}
+
+func (useCase *RecommendedPeginUseCase) getScaledProductFeePercentage() (*big.Int, error) {
+	// should be already scaled in the contract
+	uintProductFee, err := useCase.contracts.FeeCollector.DaoFeePercentage()
+	if err != nil {
+		return nil, err
+	}
+	return new(big.Int).SetUint64(uintProductFee), nil
+}
+
+func (useCase *RecommendedPeginUseCase) getGasFee(
+	ctx context.Context,
+	destinationAddress string,
+	data []byte,
+	amount *entities.Wei,
+	scaledFeePercentage *big.Int,
+) (*big.Int, error) {
+	daoFeeAmount := new(big.Int).Quo(
+		new(big.Int).Mul(amount.AsBigInt(), scaledFeePercentage),
+		big.NewInt(useCase.scale),
+	)
+	daoGasAmount, err := useCase.rpc.Rsk.EstimateGas(
+		ctx,
+		useCase.feeCollectorAddress,
+		entities.NewBigWei(daoFeeAmount),
+		make([]byte, 0),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	estimatedCallGas, err := useCase.rpc.Rsk.EstimateGas(ctx, destinationAddress, amount, data)
+	if err != nil {
+		return nil, err
+	}
+
+	gasPrice, err := useCase.rpc.Rsk.GasPrice(ctx)
+	if err != nil {
+		return nil, err
+	}
+	daoGasFee := new(big.Int).Mul(daoGasAmount.AsBigInt(), gasPrice.AsBigInt())
+	callGasFee := new(big.Int).Mul(estimatedCallGas.AsBigInt(), gasPrice.AsBigInt())
+
+	return new(big.Int).Add(callGasFee, daoGasFee), nil
+}
+
+func (useCase *RecommendedPeginUseCase) getFixedCallFee(
+	config liquidity_provider.PeginConfiguration,
+) *big.Int {
+	return config.FixedFee.AsBigInt()
+}
+
+func (useCase *RecommendedPeginUseCase) validateRecommendedValue(
+	ctx context.Context,
+	config liquidity_provider.PeginConfiguration,
+	result *big.Int,
+) error {
+	var err error
+
+	if err = config.ValidateAmount(entities.NewBigWei(result)); err != nil {
+		err = fmt.Errorf("recommended amount %s is out of range: %w", entities.NewBigWei(result).String(), err)
+		return usecases.WrapUseCaseError(usecases.RecommendedPegoutId, err)
```

Comment:
Why are you using recommended .RecommendedPegoutId here? Probably this is a copy paste error and should be .RecommendedPeginId.
Thats on this line and others in the same function.